mtwitter.js

```diff
@@ -0,0 +1,254 @@
+                          console.error("Couldn't write bearer token to file. Reason:",err);
+                      }
+                  });
+                  cb(jsonresponse.bearer_token);
+              } else {
+                  console.error("Error getting bearer token: Unexpected response");
+                  cb(null);
+              }
+          } catch (e) {
+              console.error("Error getting bearer token:", e);
+              cb(null);
+          }
+
+      }
+  });
+}
+
+mtwitter.prototype.get_user_by_id = function(id) {
+    var ts = new Date().getTime();
+    return new Promise((resolve, reject) => {
+        for (var name in this.tuser_cache) {
+            var cached = this.tuser_cache[name];
+            if (ts - cached.cache_time > TWITTER_PROFILE_INTERVAL) {
+                continue;
+            }
+            if (cached.user != null && cached.user.id_str == id) {
+                resolve(cached.user);
+            }
+        }
+        this.app_twitter.get('users/show', {
+            user_id: id
+        }, (error, user, response) => {
+            if (error) {
+                console.error(error);
+                reject(error);
+            } else {
+                this.tuser_cache[user.screen_name] = {
+                    cache_time: ts,
+                    user: user
+                };
+                resolve(user);
+            }
+        });
+    });
+}
+
+mtwitter.prototype.get_user = function(name) {
+    console.log("Looking up @" + name);
+    return new Promise((resolve, reject) => {
+        var ts = new Date().getTime();
+        if (this.tuser_cache[name] != undefined) {
+            console.log("Checking cache for @" + name);
+            var cached = this.tuser_cache[name];
+            if (ts - cached.cache_time < TWITTER_PROFILE_INTERVAL) {
+                resolve(cached.user);
+            }
+        }
+
+        console.log("Checking twitter for @" + name);
+        this.app_twitter.get('users/show', {
+            screen_name: name
+        }, (error, user, response) => {
+            if (error) {
+                this.tuser_cache[name] = {
+                    cache_time: ts,
+                    user: null
+                };
+                reject(error);
+            }
+            this.tuser_cache[name] = {
+                cache_time: ts,
+                user: user
+            };
+            resolve(user);
+        });
+    });
+
+}
+
+/*
+  Add a user's timeline to the iterator. Tweets will be automatically send to
+  the given room.
+*/
+mtwitter.prototype.enqueue_timeline = function(userid, localroom, remoteroom) {
+    var obj = {
+        "user_id": userid,
+        "local": localroom,
+        "remote": remoteroom
+    };
+    this.timeline_list.push(obj);
+    this.timeline_queue.push(obj);
+}
+
+/*
+  This function will fill the content structure for a new matrix message
+  for a given tweet.
+*/
+mtwitter.prototype.construct_message = function(tweet, type) {
```

Comment:
Good comment, poor function name. `message` is far too vague here. `get_matrix_content` perhaps?